### PR TITLE
Visualization

### PR DESCRIPTION
**Pull request by medegor44**

### CODE REVIEW

File: src/utils/spotify.js [hunk 1]
```diff
@@ -66,12 +76,20 @@ export const fetchSpotifyTracksIds = async (token, tracks) => {
   return Promise.all(tracks.map((track) => fetchSpotifyTrackId(token, track)));
 };
 
+const mapToUserModel = (spotifyModel) => {
+  return {
+    username: spotifyModel.display_name,
+    profileImage: spotifyModel.images ? spotifyModel.images[0].url : "",
```
Comment (medegor44) [inline src/utils/spotify.js]: What to do if user does not have profile image?

File: src/ResponsesView.jsx [hunk 2]
```diff
@@ -15,38 +21,67 @@ const errorType = PropTypes.shape({
 
 const responsesType = PropTypes.oneOfType([trackType, errorType]);
 
-const TrackView = ({ track }) => {
-  return <li>{track.trackId}</li>;
+const parseDuration = (durationInMs) => {
```
Comment (medegor44) [inline src/ResponsesView.jsx]: Maybe there is library or smth to transform ms to mm:ss

File: src/SpotifyUsername.jsx [hunk 3]
```diff
@@ -1,9 +1,10 @@
 import React, { useState, useEffect } from "react";
 import { fetchSpotifyUsername } from "./utils/spotify";
 import { getToken } from "./utils/tokenStorage";
+import "./css/SpotifyUsername.css";
 
 const SpotifyUsername = () => {
-  const [username, setUsername] = useState("");
+  const [user, setUser] = useState(null);
```
Comment (medegor44) [inline src/SpotifyUsername.jsx:7]: Rename SpotifyUsername into SpotifyUserProfile or smth

File: src/App.jsx [hunk 4]
```diff
@@ -25,9 +26,14 @@ const App = () => {
...
         <Route path="/">
-          <SpotifyLogin authorized={authorized} />
-          <SpotifySongSearch authorized={authorized} />
-          <SpotifyPlaylists />
+          <div className="container">
+            <div className="headContainer">
+              <h1>Spotify playlist generator</h1>
+              <SpotifyLogin authorized={authorized} />
+            </div>
+            <SpotifySongSearch authorized={authorized} />
```
Comment (munichmule) [inline src/App.jsx:34]: Use providerfor auth and other common stuff

File: src/ResponsesView.jsx (hunk 2 above)
Comment (munichmule) [reply src/ResponsesView.jsx]: momentjs or datejs come to my mind, the first one is pretty big
you may search for some tiny npm instead

File: src/utils/spotify.js (hunk 1 above)
Comment (munichmule) [reply src/utils/spotify.js]: Show some default one

File: src/ResponsesView.jsx [hunk 5]
```diff
@@ -21,26 +22,18 @@ const errorType = PropTypes.shape({
...
-      <h1 className="idx">{idx}</h1>
+      <h1 className="trackPosition">{idx + 1}</h1>
       <img className="albumCover" src={track.albumCover} alt="" />
       <div className="trackDataContainer">
         <p>{track.artist}</p>
         <p>{track.name}</p>
       </div>
       <p className="trackDataContainer">{track.album}</p>
-      <p className="trackDataContainer">{durationStr}</p>
+      <p className="trackDataContainer">{fromS(durationInSec, "mm:ss")}</p>
```
Comment (munichmule) [inline src/ResponsesView.jsx:36]: 👍

File: src/SpotifyUsername.jsx [hunk 6]
```diff
@@ -25,17 +25,26 @@ const SpotifyUsername = () => {
...
-          />
+        <div className="imageCropper">
+          {user.profileImage ? (
+            <img
+              className="profileImg"
+              src={user.profileImage}
+              alt="your profile pic"
+            />
+          ) : (
+            <h1 className="profileLetter">{user.username[0]}</h1>
```
Comment (munichmule) [inline src/SpotifyUsername.jsx:46]: 👍

**Review by munichmule — APPROVED** [review]

**Pull request merged**